executor/index_merge_reader.go

```diff
@@ -631,9 +632,12 @@ func (e *IndexMergeReaderExecutor) buildFinalTableReader(ctx context.Context, tb
 		netDataSize:      e.dataAvgRowSize * float64(len(handles)),
 	}
 	if e.isCorColInTableFilter {
+		e.tableRequestMu.Lock()
 		if tableReaderExec.dagPB.Executors, err = constructDistExec(e.ctx, e.tblPlans); err != nil {
+			e.tableRequestMu.Unlock()
 			return nil, err
 		}
+		e.tableRequestMu.Unlock()
```

Comment:
Wrapping these lines as a dedicated function seems safer and more go-style:
```
(e *IndexMergeReaderExecutor) constructDistExec() {
    e.Lock()
    defer e.Unlock()
    constructDistExec(e.ctx, e.tblPlans)
}
```

Reply:
Have fixed.